controllers/terminal_controller.go

```diff
@@ -1327,6 +1335,43 @@ func NewClientSetFromServiceAccountRef(ctx context.Context, cs *ClientSet, ref *
 	})
 }
 
+func NewClientSetFromShootRef(ctx context.Context, cs *ClientSet, ref *extensionsv1alpha1.ShootRef, scheme *runtime.Scheme) (*ClientSet, error) {
```

Comment:
Why not move all the code related to client creation into an own package and globally initialize the scheme there.